### PR TITLE
Add suggest_form_store toolbar action for side-by-side AI suggestions

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Introduces `sulu_admin.suggest_form_store`, a new form toolbar action that is a sibling of `update_form_store` for toolbar actions that generate content over an AI endpoint. If the target fields are still empty, it generates and applies directly, same as `update_form_store` and without the optimize form (there is nothing to optimize yet). If they already hold content, a comparison dialog opens as soon as it is clicked, generates immediately, and shows the suggestion next to the current content with a regenerate button and an editable optimize checkbox, applying only once the user confirms with Insert.

Both comparison columns render through the same `suggestionFormKey` schema. Read-only is not automatic: every property in that schema must set `disabledCondition="true"` for the columns to stay non-editable, same as any other Sulu form field. Column headings and the regenerate button label default to translated strings (`sulu_admin.current_content`, `sulu_admin.suggestion`, `sulu_admin.regenerate`) and can be overridden per consumer via `originalColumnLabel`, `suggestionColumnLabel` and `regenerateText`.

The transport and dialog plumbing the two toolbar actions share (URL building, jexl content expressions, retry/error handling) is factored out into a new `AbstractGenerateFormStoreToolbarAction` base class. `UpdateFormStoreToolbarAction`'s own behaviour and its existing test suite are unchanged.

#### Why?

`UpdateFormStoreToolbarAction`'s confirm dialog is already reused by some consumers as a plain input form (via a no-op `contentExpressions` entry), so folding a side-by-side comparison flow into that same class would mix two incompatible meanings of its dialog. A sibling action keeps both use cases clean.

#### Example Usage

```php
new ToolbarAction('sulu_admin.suggest_form_store', [
    'icon' => 'su-magic',
    'label' => 'Generate',
    'route' => 'my_generate_route',
    'dialogKey' => 'my-generate-dialog',
    'dialogTitle' => 'Overwrite current content?',
    'dialogDescription' => 'Existing content will be shown next to the suggestion.',
    'dialogOkText' => 'Generate',
    'dialogCancelText' => 'Cancel',
    'dialogInsertText' => 'Insert',
    'suggestionFormKey' => 'my_suggestion_form',
    'originalColumnLabel' => 'Current Content',
    'suggestionColumnLabel' => 'Suggestion',
    'regenerateText' => 'Regenerate',
    'contentExpressions' => [
        ['property' => 'title', 'get' => 'title', 'path' => '/title'],
    ],
]);
```

`my_suggestion_form` must mark every property `disabledCondition="true"` to keep the comparison read-only:

```xml
<property name="title" type="text_line" disabledCondition="true">
    <meta>
        <title>Title</title>
    </meta>
</property>
```

#### To Do

- [ ] Create a documentation PR
- [ ] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+    destroy() {
+        this.destroySuggestionStores();
+    }
+
+    getNode() {
+        return (
+            <Dialog
+                cancelText={this.dialogCancelText || translate('sulu_admin.cancel')}
+                confirmDisabled={this.loading || !this.suggestionFormStore}
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Insert can be clicked when there is no suggestion, and doing so blanks the target fields. Both ways to get there are reproduced in the browser:

1. **After a temporary error.** The catch sets `loading = false` and keeps the dialog open, so Insert is enabled over an empty suggestion. Clicking it runs `writeContentExpressions()` with `undefined` for every property. The page title and article were both blank afterwards, with the form dirty and one save away from persisting it.
2. **While the suggestion schema is still loading** (also flagged by Codex on this line). I reproduced it by delaying `/admin/metadata/form/<suggestionFormKey>` and answering the generate request quickly. Insert was enabled over two loaders, and clicking it blanked the title.

Insert should only be enabled once a response has actually reached a loaded suggestion store, for example:

```js
@observable hasSuggestion = false;

// generate(), before the request
this.hasSuggestion = false;

// .then(), after changeMultiple()
this.hasSuggestion = true;

confirmDisabled={this.loading || !this.hasSuggestion || !!this.suggestionFormStore?.loading}
```

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        this.dialogSnackbarMessage = undefined;
+        this.loading = true;
+        this.blankSuggestionFields();
+
+        const url = this.buildRequestUrl();
+        const content = await this.getCurrentContent();
+
+        Requester.post(url, {
+            content,
+            data: this.formStore?.data || {},
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Reproduced in the browser: on the first open after loading the admin, the request is sent with `data: {}`. In my run the POST went out 31 ms after the click, and the metadata for the options form answered at 115 ms. So the options store only held `{optimize: true}` after the request was already on its way. The dialog then shows Optimize ticked next to a suggestion that was generated without it. The second open in the same page load sends `{optimize: true}`, because the metadata is cached by then.

Waiting for the options store before building the body fixes it. Without an effect, mobx's `when` returns a promise:

```js
const content = await this.getCurrentContent();
const formStore = this.formStore;
if (formStore) {
    await when(() => !formStore.loading);
}

Requester.post(url, {
    content,
    data: formStore?.data || {},
})
```

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestionColumn.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export default class SuggestionColumn extends React.Component<Props> {
+    render(): Node {
+        const {children, heading, store} = this.props;
+
+        return (
+            <div className={suggestionColumnStyles.column}>
+                {heading && <div className={suggestionColumnStyles.heading}>{heading}</div>}
+
+                <div className={suggestionColumnStyles.card}>
+                    {store && <FormContainer onSubmit={noop} store={store} />}
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestionColumn.js:33]: Both columns are editable: both title inputs accept input, and both CKEditors are `contenteditable`. Whatever is typed into the suggestion column is what Insert writes; I inserted a hand-edited title that way. Whatever is typed into the current-content column is silently thrown away.

Which behaviour do you want? If editing the suggestion before inserting is a feature, I'd keep it, say so in the description, and make only the current-content column read-only. If both columns should be read-only as the description says, note that `FormContainer` has no read-only mode today. Fields only get disabled through `disabledCondition` in the schema, so the container would need a way to force its fields disabled.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                key={this.dialogKey}
+                onCancel={this.handleDialogClose}
+                onConfirm={this.handleConfirm}
+                open={this.showDialog}
+                size={this.originalFormStore ? 'large' : undefined}
+                snackbarMessage={this.dialogSnackbarMessage}
+                snackbarType={this.dialogSnackbarType}
+                title={this.dialogTitle}
+            >
+                {Boolean(this.originalFormStore) && (
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:309]: The dialog never renders `dialogDescription`, although `AbstractGenerateFormStoreToolbarAction` throws without it. With the example from the description, the text "Existing content will be shown next to the suggestion." never appears. Either render it above the grid, as `UpdateFormStoreToolbarAction` does, or drop it from the required options for this action.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+jest.mock('../../../../utils/Translator', () => ({
+    translate: jest.fn((key) => key),
+}));
+
+jest.mock('../../../../containers/Form/stores/metadataStore', () => ({
+    getSchema: jest.fn().mockReturnValue(Promise.resolve({})),
+    getJsonSchema: jest.fn().mockReturnValue(Promise.resolve({})),
+}));
+
+jest.mock('../../../../containers/Form/stores/memoryFormStoreFactory', () => ({
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js:34]: This mock returns stores whose `data` is available immediately, which is exactly what hides #1 and #2. The test "Open the dialog and start generating immediately when content exists, with optimize defaulted on" expects `data: {optimize: true}` and passes, while the browser sends `data: {}`. The `metadataStore` mock above it is never reached, because the factory itself is mocked.

To check, I turned single lines into no-ops, one at a time. All 11 tests still pass with each of these:
- `confirmDisabled={false}`
- no `this.loading = true` in `generate()`
- no `this.dialogSnackbarType = 'warning'`

Could one test use the real `memoryFormStoreFactory` with a `metadataStore` mock that resolves later? It could then check the request body, and that Insert is disabled while generating, after a temporary error, and before the suggestion store is ready.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        return suggestionFormKey;
+    }
+
+    @computed get insertText() {
+        const {dialogInsertText} = this.options;
+
+        return typeof dialogInsertText === 'string' ? dialogInsertText : this.dialogOkText;
+    }
+
+    @computed get originalColumnLabel() {
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:83]: The configuration from the PR description sets none of `originalColumnLabel`, `suggestionColumnLabel` or `regenerateText`, and they have no defaults. The dialog then shows two identical "Title" columns with nothing saying which one is the suggestion. The regenerate button reads "Generate", the same word as the toolbar button. Could these default to translations (e.g. `sulu_admin.current_content`, `sulu_admin.suggestion`, `sulu_admin.regenerate`)? They should also be listed in the description and in the docs PR.

**Review by Prokyonn — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in 0b59e4e6. Added a `hasSuggestion` flag that only turns true once a response has landed in a loaded suggestion store. `confirmDisabled` now checks `hasSuggestion` and `suggestionFormStore.loading` together. Verified live: Insert now renders disabled after a temporary error and while the store is still loading.

Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in 0b59e4e6, same fix as noted in the Copilot P1 comment above: a `hasSuggestion` flag gates `confirmDisabled`, checked alongside `suggestionFormStore.loading`. Verified live with Playwright: Insert now renders `disabled` and refuses the click after a temporary error.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+                        formMetadataOptions
+                    );
+                }
+
+                this.originalFormStore = memoryFormStoreFactory.createFromFormKey(this.suggestionFormKey, contentData);
+                this.suggestionFormStore = memoryFormStoreFactory.createFromFormKey(this.suggestionFormKey, {});
+                this.openDialog();
+            })();
+
+            this.generate();
```
[the remark this answers is not in the document]
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:126]: Fixed in 0b59e4e6. `generate()` now awaits the optimize form store's own loading state (`await when(() => !formStore.loading)`) before reading its data, so the request body always reflects the resolved schema, not the pre-load default.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 2 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in 0b59e4e6, same race as Copilot's comment on line 122: `generate()` now awaits the optimize form store's loading state before building the request body. Added a test using the real `memoryFormStoreFactory` with a controllable `metadataStore` mock to reproduce the timing.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestionColumn.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestionColumn.js:33]: Checked live: with the shipped `suggestionFormKey` schemas (`disabledCondition="true"` on every property), both columns already render as genuinely disabled inputs - confirmed both visually and by Playwright refusing to fill them ("element is not enabled"). Documented the contract in the class docblock and the PR description, since read-only depends on the schema author setting `disabledCondition`, not something this component enforces on its own.

Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestionColumn.js:33]: Reproduced this live first to be sure, then found it does not reproduce with the shipped suggestion form schemas: every property sets `disabledCondition="true"`, and Playwright confirms the resulting inputs are genuinely disabled (fill attempts fail with "element is not enabled"). Went with documenting the contract instead of adding a new core read-only API for one experimental action - noted in the class docblock and the PR description that `suggestionFormKey` schemas must set `disabledCondition` on every property.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 4 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:309]: Fixed in 0b59e4e6, matches Copilot's P2 comment: `dialogDescription` now renders above the comparison grid, same placement as `UpdateFormStoreToolbarAction`.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js (hunk 5 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js:34]: Fixed in 0b59e4e6. Added tests asserting on `hasSuggestion`/`loading` directly plus one using the real `memoryFormStoreFactory` with a controllable `metadataStore` mock for the race condition. Re-ran your three mutations (`confirmDisabled={false}`, dropped `this.loading = true`, dropped `dialogSnackbarType = 'warning'`) against the new suite - each now fails a test, then reverted.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 6 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:83]: Fixed in 0b59e4e6. `originalColumnLabel`, `suggestionColumnLabel`, `regenerateText` and `insertText` now fall back to new translations (`sulu_admin.current_content`, `sulu_admin.suggestion`, `sulu_admin.regenerate`, existing `sulu_admin.insert`) instead of duplicating other buttons' text. Also listed the three label options in the PR description and the bundle docs.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+                onCancel={this.handleDialogClose}
+                onConfirm={this.handleConfirm}
+                open={this.showDialog}
+                size={this.originalFormStore ? 'large' : undefined}
+                snackbarMessage={this.dialogSnackbarMessage}
+                snackbarType={this.dialogSnackbarType}
+                title={this.dialogTitle}
+            >
+                {Boolean(this.originalFormStore) && (
+                    <Grid>
```
[the remark this answers is not in the document]
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:310]: Fixed in 0b59e4e6. `dialogDescription` now renders above the comparison grid, same placement as `UpdateFormStoreToolbarAction`.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+        this.suggestionFormStore?.changeMultiple(
+            Object.keys(data).reduce((values, key) => ({...values, [key]: undefined}), {}),
+            {isDefaultValue: true}
+        );
+    }
+
+    writeContentExpressions(getValue: (property: string) => mixed) {
+        for (const expr of this.contentExpressions) {
+            if (expr.path) {
+                this.resourceFormStore.change(expr.path, getValue(expr.property));
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: When the endpoint answers with only some of the configured properties, Insert writes `undefined` into the others and deletes existing content.

With `contentExpressions` for title and article and an answer of `{"title": "..."}`, the suggestion column shows the new title and an empty Article box, which reads like "nothing generated for this field". Insert then writes `undefined` over the article that was there. I reproduced it in the admin: one click and the page's article was gone.

Could `writeContentExpressions` skip the properties the response does not carry, so an incomplete answer can only add, never delete?

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js [hunk 10]
```diff
@@ -0,0 +1,505 @@
...
+            formKey
+        )
+    );
+
+    const config = action.getToolbarItemConfig();
+    await config.onClick();
+
+    // formStore's schema has not resolved yet - the request must not have gone out with a
+    // premature, still-loading "data" value
+    expect(Requester.post).not.toHaveBeenCalled();
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js:500]: This assertion cannot fail, so the test passes even without the fix it covers. Replacing `await when(() => !formStore.loading)` with a no-op leaves all 14 tests green.

`await config.onClick()` returns before `generate()` has finished its own `await this.getCurrentContent()`, so no request could have been sent at this point either way, and `resolveSchema({})` on the next line runs before the request body is built. One line after `await config.onClick();` fixes it:

```js
await new Promise((resolve) => setTimeout(resolve));
```

With that line the test stays green on the current code and fails once the guard is removed ("Expected number of calls: 0, Received: 1").

Two more single-line no-ops that no test catches: removing `this.hasSuggestion = false` in `generate()`, and removing `{this.dialogDescription}` from the dialog. The first one matters, since it is what keeps Insert disabled when a regenerate fails after a suggestion had already loaded. A test for "suggestion loaded, regenerate answers a temporary error, Insert disabled" would pin it.

Unrelated to the assertion, the comment on line 322 says the dialog stays open over the previous suggestion, but `generate()` blanks the fields first, so what stays on screen is empty.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+        this.blankSuggestionFields();
+
+        const url = this.buildRequestUrl();
+        const content = await this.getCurrentContent();
+        const formStore = this.formStore;
+
+        // the optimize checkbox's schema loads asynchronously too - without this, a request
+        // fired before it resolves goes out with formStore.data still at its pre-load default
+        if (formStore) {
+            await when(() => !formStore.loading);
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: When the options form's own schema request fails, this wait never ends: `SchemaFormStoreDecorator` has no error state, so its `loading` stays true.

I let `/admin/metadata/form/<formKey>` answer 500 in the admin: no generate request went out at all, the dialog sat there with spinners and no message, and after Cancel the toolbar button stayed busy until I reloaded the page. Before this commit the same failure still generated, just without the options.

Worth giving up when the schema request fails and falling through to the generic error, so the user sees something.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+    @action generateAndApply = async() => {
+        this.clearRetryWarning();
+        this.loading = true;
+
+        const url = this.buildRequestUrl();
+        const content = await this.getCurrentContent();
+
+        Requester.post(url, {
+            content,
+            data: {},
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:182]: When the target fields are empty, this path posts a hardcoded empty `data`, so the options form is skipped even when the action is configured with `formKey`. The same action then generates with `{"optimize": true}` through the dialog and without it here, which the endpoint sees as two different requests.

If that is intended, the PR description could say so: it currently says the dialog opens as soon as the action is clicked, which only holds when the fields already have content.

**Review by Prokyonn — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 9 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in bf5eecaa. `writeContentExpressions` now skips properties whose value is still `undefined` (i.e. the response/suggestion never answered them), instead of writing `undefined` over content that was already there. Added a test with a partial response (title answered, article not) asserting the untouched field is never written.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js (hunk 10 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/SuggestFormStoreToolbarAction.test.js:500]: Right, that assertion couldn't fail as written. Fixed in bf5eecaa: added `await new Promise((resolve) => setTimeout(resolve));` right after `await config.onClick();`, exactly as suggested, so `generate()` actually reaches its own await before the check runs. Verified it now fails with "Expected number of calls: 0, Received: 1" when the wait is removed, then reverted.

Added the "suggestion loaded, regenerate answers a temporary error, Insert disabled" test too - verified it fails when `this.hasSuggestion = false` is dropped from `generate()`, then reverted.

Also fixed the comment on the temporary-error test: you're right that `generate()` blanks the fields first, so what's on screen is empty, not a stale suggestion.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 11 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in bf5eecaa. Added a `waitForFormStoreToLoad()` helper that races the `when()` wait against a 10s timeout; on timeout it falls through to the same temporary-error warning branch as any other failure (dialog stays open, loading resets, Regenerate is the retry). Added a test using fake timers that simulates a formStore stuck in `loading: true` and asserts the fallback fires.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 12 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js:182]: This is intentional and matches `UpdateFormStoreToolbarAction`'s own `handleClick`, which has the identical asymmetry (the options form is only created inside its `hasExistingContent` branch too) - there's nothing to optimize when the fields are still empty, so the flag doesn't semantically apply. Updated the PR description to say so explicitly instead of implying the dialog always opens immediately.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 11 above)
Comment (Prokyonn) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: The timeout stops the hang, but Regenerate cannot recover after it fires. The options store in this dialog is the same `SchemaFormStoreDecorator` whose schema request failed, and it never asks again, so every Regenerate click waits another 10 seconds and shows the same warning, even once the metadata endpoint answers normally. I checked it in the admin: no request went out on Regenerate, while Cancel and reopening recovered straight away, because the failed metadata request is dropped from the cache and the new store asks for it again.

Either recreate the options store before returning here, so the next Regenerate requests the schema again, or close the dialog with the error so the user reopens it, which already works.

**Review by Prokyonn — APPROVED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js (hunk 11 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SuggestFormStoreToolbarAction.js]: Fixed in 40f507ad48. `generate()` now destroys the stuck `formStore` on timeout and recreates it from the same `formMetadataOptions`, so the next Regenerate click asks for the schema again instead of hitting the same dead store.
